scipy/stats/_mstats_extras.py

```diff
@@ -165,10 +165,14 @@ def _hdsd_1D(data, prob):
         for (i,p) in enumerate(prob):
             _w = betacdf(vv, n*p, n*(1-p))
             w = _w[1:] - _w[:-1]
-            mx_ = np.fromiter([w[:k] @ xsorted[:k] + w[k:] @ xsorted[k+1:]
-                               for k in range(n)], dtype=float_)
-            # mx_var = np.array(mx_.var(), copy=False, ndmin=1) * n / (n - 1)
-            # hdsd[i] = (n - 1) * np.sqrt(mx_var / n)
+            # cumulative sum of weights and data points if
+            # ith point is left out for jackknife
+            cumulativeSumLeft = np.append([0], np.cumsum(w * xsorted[:-1]))
+            # similar but from the right
+            cumulativeSumRight = np.append([0],
+                                           np.cumsum(w[::-1] * xsorted[:0:-1])
+                                           )[::-1]
+            mx_ = cumulativeSumLeft + cumulativeSumRight
```

Comment:
This is good. Would you consider the following?
```suggestion
            mx_ = np.zeros_like(xsorted)
            mx_[1:] = np.cumsum(w * xsorted[:-1])
            # similar but from the right
            mx_[:-1] += np.cumsum(w[::-1] * xsorted[:0:-1])[::-1]
```
It's a bit more concise since it avoids the appends. Of course, we could get fancier (e.g. `np.empty_like` instead of `np.zeros_like`, use the `out` argument of `cumsum` to perform the calculation in `mx_` directly) but I think this is a good compromise between readability and performance.

Reply:
Yeah this looks good.